api/v1beta1/foundationdbcluster_types.go

```diff
@@ -1187,6 +1187,15 @@ type FoundationDBClusterAutomationOptions struct {
 	// +kubebuilder:validation:Enum=All;Zone;ProcessGroup
 	// +kubebuilder:default:=Zone
 	DeletionMode DeletionMode `json:"deletionMode,omitempty"`
+
+	// RemovalMode defines the removal mode for this cluster. This can be
+	// DeletionModeAll, DeletionModeZone or DeletionModeProcessGroup. The
+	// RemovalMode defines how process groups are deleted in order when they
+	// are marked for removal.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Enum=All;Zone;ProcessGroup
+	// +kubebuilder:default:=Zone
+	RemovalMode DeletionMode `json:"removalMode,omitempty"`
```

Comment:
Actually `DeletionMode` is perhaps not the best name but I didn't came up with a better name.

Reply:
I like the idea of renaming the type to `UpdateMode`, and coming up with a better name for the current `DeletionMode` field on the spec. Maybe `PodUpdateMode` would cover it, since there's a general understanding that "updating" a pod involves deleting and recreating it, and `RecreationMode` doesn't look right.